core/embed/sys/mpu/stm32u5/mpu.c

```diff
@@ -131,20 +131,32 @@ static void mpu_set_attributes(void) {
 _Static_assert(NORCOW_SECTOR_SIZE == STORAGE_1_MAXSIZE, "norcow misconfigured");
 _Static_assert(NORCOW_SECTOR_SIZE == STORAGE_2_MAXSIZE, "norcow misconfigured");
 
-// PERIPH_SIZE covers secure peripherals only (+16MB of FMC1)
-// PERIPH_SIZE_EXT covers both secure and non-secure peripherals (+16MB of FMC1)
-// The extended size is used in a special case - MPU_MODE_OTP - when access
-// to non-secure FLASH controller registers is required.
-
-#ifdef STM32U585xx
-// On STM32U585, we need to add an additional 16M for FMC1 which
+// PERIPH_SIZE covers both secure and non-secure peripherals
+// 0x40000000 to 0x4FFFFFFF (256M) and
+// 0x50000000 to 0x5FFFFFFF (256M).
+
+// When writing to OTP memory while running in secure mode, we have to access
+// the non-secure FLASH peripheral. Additionally, the ST HAL requires access
+// to the same non-secure peripheral during the *first* write or erase
+// peration after writing to OTP. This applies even if the following operation
+// targets a different flash region. To avoid faults, we must ensure
+// the MPU permanently allows access to non-secure peripherals.
+
+// Moreover, on STM32U585, we need to add an additional 16M for FMC1 which
 // follows the peripherals in the memory map.
+// 0x60000000 to 0x60FFFFFF (16M).
 
-#define PERIPH_SIZE (SIZE_256M + SIZE_16M)
-#define PERIPH_SIZE_EXT (SIZE_512M + SIZE_16M)
+// In the kernel, on models with a secure monitor (SECURE_MODE is not defined),
+// we can allow access *only* to the non-secure peripherals region.
+
+#ifdef STM32U585xx
```

Comment:
it doesn't matter now but consider reversing order of the ifdefs: decide the MCU type insude the SECURE_MODE if, that way, it will be compatible when we enable non-secure mode on U585

Reply:
ah ok, didn't realize that